llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -3597,10 +3597,22 @@ bool AArch64InstructionSelector::selectBrJT(MachineInstr &I,
   unsigned JTI = I.getOperand(1).getIndex();
   Register Index = I.getOperand(2).getReg();
 
+  MF->getInfo<AArch64FunctionInfo>()->setJumpTableEntryInfo(JTI, 4, nullptr);
+  if (MF->getFunction().hasFnAttribute("jump-table-hardening") ||
+      STI.getTargetTriple().isArm64e()) {
+    if (TM.getCodeModel() != CodeModel::Small)
```

Comment:
Do we also need one for `Tiny`? Not yet supported for Mach-O, but it would be a nasty surprise otherwise.

Reply:
I agree that if we check against `Small`, we might also want to check against `Tiny`. I'm actually not sure if such a check is essential here at all - see https://github.com/llvm/llvm-project/pull/97666#discussion_r1668053989

Reply:
I misread. This is already asserting that only `Small` is supported. Support for other models can come in a later patch as needed.